fmralign/alignment_methods.py

```diff
@@ -465,3 +469,266 @@ def fit(self, X, Y):
+        self.n_components = n_components
+        self.n_jobs = n_jobs
+
+    ################################################################
+    # Computing decomposition
+
+    @staticmethod
+    def _tuning_estimator(shared_response, target):
+        """
+        Estimate the tuning matrix for individualized neural tuning.
+
+        Parameters:
+        ----------
+        shared_response : array-like
+             The shared response matrix of shape (n_timepoints, k)
+             where k is the dimension of the sources latent space.
+        target : array-like
+             The target matrix.
+        latent_dim : int, optional
+             The number of latent dimensions (if PCA is used). Defaults to None.
+
+        Returns:
+        --------
+        array-like: The estimated tuning matrix for the given target.
+
+        """
+        if shared_response.shape[1] != shared_response.shape[0]:
+            return (np.linalg.pinv(shared_response)).dot(target)
+        return np.linalg.inv(shared_response).dot(target)
+
+    @staticmethod
+    def _stimulus_estimator(full_signal, n_subjects, latent_dim=None, scaling=True):
+        """
+        Estimates the stimulus matrix for the Individualized Neural Tuning model.
+
+        Parameters:
+        -----------
+        full_signal : numpy.ndarray
+             Concatenated signal for all subjects,
+             of shape (n_timepoints, n_subjects * n_voxels).
+        n_subjects : int
+             The number of subjects.
+        latent_dim : int, optional
+             The number of latent dimensions to use. Defaults to None.
+        scaling : bool, optional
+             Whether to scale the stimulus matrix sources. Defaults to True.
+        """
+        n_timepoints = full_signal.shape[0]
+        if scaling:
+            U = svd_pca(full_signal)
+        else:
+            U, _, _ = safe_svd(full_signal)
+        if latent_dim is not None and latent_dim < n_timepoints:
+            U = U[:, :latent_dim]
+
+        stimulus = np.sqrt(n_subjects) * U
+        stimulus = stimulus.astype(np.float32)
+        return stimulus
+
+    @staticmethod
+    def _reconstruct_signal(shared_response, individual_tuning):
+        """
+        Reconstructs the signal using the shared response and individual tuning.
+
+        Parameters:
+        --------
+        shared_response : numpy.ndarray
+             The shared response of shape (n_timeframes, n_timeframes) or
+             (n_timeframes, latent_dim).
+        individual_tuning : numpy.ndarray
+             The individual tuning of shape (latent_dim, n_voxels) or
+             (n_timeframes, n_voxels).
+
+        Returns:
+        --------
+        ndarray: The reconstructed signal of shape (n_timeframes, n_voxels)
+        (same shape as the original signal)
+        """
+        return (shared_response @ individual_tuning).astype(np.float32)
+
+    def fit(
+        self,
+        X,
+        searchlights=None,
```

Comment:
All these arguments should be given to the constructor, not to the fit method. Ideally, it only has the X and verbose arguments.

Reply:
Well, rewriting the core of the code the last day seems a bit dangerous for me, as a lot of other code depends on it. Will try anyway

Reply:
Edit : done !